src/datachain/func/string.py

```diff
@@ -152,3 +152,49 @@ def inner(arg):
         args = None
 
     return Func("regexp_replace", inner=inner, cols=cols, args=args, result_type=str)
+
+
+def byte_hamming_distance(*args: Union[str, Func]) -> Func:
+    """
+    Computes the Hamming distance between two strings.
+
+    The Hamming distance is the number of positions at which the corresponding
+    characters are different. This function returns the dissimilarity between
+    the strings, where 0 indicates identical strings and values closer to the length
+    of the strings indicate higher dissimilarity.
+
+    Args:
+        args (str | literal): Two strings to compute the Hamming distance between.
+            If a string is provided, it is assumed to be the name of the column.
+            If a literal is provided, it is assumed to be a string literal.
```

Comment:
I don't understand the difference between string (column) and literal (string). Can you find a better wording?

Reply:
Hm... This is a good example: https://github.com/iterative/datachain/pull/669/files#diff-1a98b3b24a2c496f079d2c5f5757a55850ca5840a7e53350e39c5d1c10175979R640
but let me think how to update the docstrings.

Reply:
Oh! Maybe I was too picky; now I understand it.

Reply:
No, you're right, it should be clear. May be if I'll update an example below: https://github.com/iterative/datachain/pull/669/files/27108dc1862f92f500482fd1f2457bbdaf10002d#diff-cd56a8e471c52456b3e8e5492b585f075a6534b01d2fcc3af4a82b7ead109323R177 it will be enough?

Reply:
Like this?
```python
        dc.mutate(
            ham_dist=func.byte_hamming_distance("embed1", literal("hello")),
        )
```

Reply:
For the sake of comparing pears with pears, what if you do this?

```diff
- If a string is provided, it is assumed to be the name of the column.
+ If a str is provided, it is assumed to be the name of the column.
- If a literal is provided, it is assumed to be a string literal.
+ If a Literal is provided, it is assumed to be a string literal.
```

I also tend to prefer full words over abbreviations, but in this case using the `str` abbreviation and the title case `Literal` makes clear we're referring to Python/SQLAlchemy types, and not just vague concepts.

Reply:
And yes, the new example is awesome, thanks! 🤩